### PR TITLE
Fixes #29702 - added image validation check to ovirt

### PR DESCRIPTION
**Pull request by yifatmakias**

### CODE REVIEW

File: app/controllers/api/v2/images_controller.rb [hunk 1]
```diff
@@ -48,6 +48,7 @@ def show
       param_group :image, :as => :create
 
       def create
+        validate_params unless Rails.env.test?
```
Comment (timogoebel) [inline app/controllers/api/v2/images_controller.rb]: I believe we should not skip this in the test environment. Please stub the method in tests if you want to skip it.

Comment (yifatmakias) [reply app/controllers/api/v2/images_controller.rb]: I fixed the problem in a better way, no need for stubs at all now.

File: app/models/compute_resources/foreman/model/ovirt.rb [hunk 2]
```diff
@@ -138,6 +138,13 @@ def available_images
       templates
     end
 
+    def image_exists?(image)
+      client.templates.get(image)
+      true
```
Comment (kgaikwad) [inline app/models/compute_resources/foreman/model/ovirt.rb]:
```suggestion
       client.templates.get(image).present?
```

File: app/models/compute_resources/foreman/model/ovirt.rb [hunk 3]
```diff
@@ -138,6 +138,13 @@ def available_images
...
+    rescue OvirtSDK4::NotFoundError
+      false
```
Comment (kgaikwad) [inline app/models/compute_resources/foreman/model/ovirt.rb:145]: It would be good to log exception using `Foreman::Logging.exception`.
What do you think?

**Review by kgaikwad — CHANGES_REQUESTED** [review]
@yifatmakias, added small inline comments.

File: app/models/compute_resources/foreman/model/ovirt.rb (hunk 3 above)
Comment (yifatmakias) [reply app/models/compute_resources/foreman/model/ovirt.rb:145]: I agree I will add it.

**Review by kgaikwad — APPROVED** [review]
I haven't tested yet but code wise looks good to me.
Thank you @yifatmakias.

@timogoebel, any additional comments?

**Comment by timogoebel** [comment]
I‘m good, thanks!

**Comment by shiramax** [comment]
@kgaikwad works fine! you can merge

**Comment by kgaikwad** [comment]
> @kgaikwad works fine! you can merge

+1 thank you @shiramax for helping me in testing. Merging the pull-request.

**Comment by kgaikwad** [comment]
Thank you @yifatmakias and @timogoebel

**Pull request merged**